src/EoS/Nuclear/NuclearEoS.cpp

```diff
@@ -0,0 +1,210 @@
+//                alltables_mode : Auxiliary log(eps) arrays for different modes
+//                logrho         : log(rho) array in the table
+//                logeps         : log(eps) array in the table
+//                yes            : Y_e      array in the table
+//                logtemp_mode   : log(T)   array for the temperature mode
+//                entr_mode      : entropy  array for the entropy mode
+//                logprss_mode   : log(P)   array for the pressure mode
+//                keymode        : Which mode we will use
+//                                 --> 0 : Energy mode      (input eps    )
+//                                     1 : Temperature mode (input T      )
+//                                     2 : Entropy mode     (input entropy)
+//                                     3 : Pressure mode    (input P      )
+//                keyerr         : Output error
+//                                 --> 667 : fail in finding energy (T, e, P modes)
+//                                     101 : Y_e too high
+//                                     102 : Y_e too low
+//                                     103 : eps too high (if keymode = 0)
+//                                     104 : eps too low  (if keymode = 0)
+//                                     105 : rho too high
+//                                     106 : rho too low
+//                rfeps          : Tolerence for interpolations
+//                                 --> Not used currently
+//-----------------------------------------------------------------------------------------------
+GPU_DEVICE
+void nuc_eos_C_short( const real xrho, real *xenr, const real xye,
+                      real *xtemp, real *xent, real *xprs,
+                      real *xcs2, real *xmunu, const real energy_shift,
+                      const int nrho, const int neps, const int nye, const int nmode,
+                      const real *alltables, const real *alltables_mode,
+                      const real *logrho, const real *logeps, const real *yes,
+                      const real *logtemp_mode, const real *entr_mode, const real *logprss_mode,
+                      const int keymode, int *keyerr, const real rfeps )
+{
+
+   real lr   = LOG10( xrho );
+   real leps = ( keymode == NUC_MODE_ENGY ) ? LOG10( *xenr + energy_shift ) : NULL_REAL;
+
+
+// check whether (rho, eps, Y_e) is within the table
+   *keyerr = 0;
+
+   if ( lr > logrho[nrho-1] )
+   {
+      *keyerr = 105;
+      return;
+   }
+
+   if ( lr < logrho[0] )
+   {
+      *keyerr = 106;
+      return;
+   }
+
+   if ( xye > yes[nye-1] )
+   {
+      *keyerr = 101;
+      return;
+   }
+
+   if ( xye < yes[0] )
+   {
+      *keyerr = 102;
+      return;
+   }
+
+   if ( keymode == NUC_MODE_ENGY )
+   {
+      if ( leps > logeps[neps-1] )
+      {
+         *keyerr = 103;
+         return;
+      }
+
+      if ( leps < logeps[0] )
+      {
+         *keyerr = 104;
+         return;
+      }
+   }
+
+
```

Comment:
We'd better to check if `xrho`, `xeps`, `temp`, `xent`, `xprs` and `xye` are in the range of our table at the beginning of the function `nuc_eos_C_short(...)`. Then we do not need to check this condition later in  `findenergy(...)` and `nuc_eos_C_cubinterp_some(...)`.
```suggestion
   if ( keymode == NUC_MODE_TEMP )
   {
      if ( LOG10( *xtemp ) > logtemp_mode[nmode-1] )
      {
         *keyerr = 107;
         return;
      }

      if ( LOG10( *xtemp ) < logtemp_mode[0] )
      {
         *keyerr = 108;
         return;
      }
   }
   
      if ( keymode == NUC_MODE_ENTR )
   {
      if ( *xent > entr_mode[nmode-1] )
      {
         *keyerr = 109;
         return;
      }

      if ( *xent < entr_mode[0] )
      {
         *keyerr = 110;
         return;
      }
   }

   if ( keymode == NUC_MODE_PRES )
   {
      if ( LOG10( *xprs ) > logprss_mode[nmode-1] )
      {
         *keyerr = 111;
         return;
      }

      if ( LOG10( *xprs ) < logprss_mode[0] )
      {
         *keyerr = 112;
         return;
      }
   }
```

Reply:
Done